app/src/openxr/cpp/OpenXRInputSource.cpp

```diff
@@ -1069,19 +1071,27 @@ OpenXRInputSource::GetNextHandMeshBuffer() {
     return mHandMeshMSFT.buffer;
 }
 
-void OpenXRInputSource::HandleEyeTrackingScroll(XrTime predictedDisplayTime, bool triggerClicked, const vrb::Matrix &pointerTransform, ControllerDelegate &controllerDelegate) {
+bool OpenXRInputSource::HandleEyeTrackingScroll(XrTime predictedDisplayTime, bool triggerClicked, const vrb::Matrix &pointerTransform, const vrb::Matrix& eyeTrackingTransform, ControllerDelegate &controllerDelegate) {
+    bool handled { false };
     if (!mTriggerWasClicked && triggerClicked) {
-        mControllerPositionOnGestureStart =  pointerTransform.GetTranslation();
+        mEyeGazeTransformOnPinchStart = eyeTrackingTransform;
         mEyeTrackingPinchStartTime = predictedDisplayTime;
+        handled = true;
     } else if (mTriggerWasClicked && triggerClicked) {
         // Throttle the start of the scroll gesture to avoid scrolling on pinch.
         if (predictedDisplayTime - mEyeTrackingPinchStartTime > kEyeTrackingScrollThreshold) {
             vrb::Vector currentControllerPosition = pointerTransform.GetTranslation();
-            auto delta = currentControllerPosition - mControllerPositionOnGestureStart;
-            controllerDelegate.TranslateTransform(mIndex, delta * 10);
+            auto delta = currentControllerPosition - mControllerPositionOnScrollStart;
+            controllerDelegate.SetTransform(mIndex, mEyeGazeTransformOnPinchStart.Translate(delta * 10));
+        } else {
+            // Keep updating the initial position while throttling to avoid a sudden jump when
+            // scrolling starts.
+            mControllerPositionOnScrollStart =  pointerTransform.GetTranslation();
         }
+        handled = true;
```

Comment:
If I've understood correctly how  'handled' is updated,  we could  just return the value of triggerCliced, right ? After all, we only use it as the function's return value.

That makes  me think why we need the function to return any value, since the triggerClicked value comes form an argument of the function.

Reply:
Hmm that's a good point, I'll check it out